x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/pages/coverage_overview/reducer.ts

```diff
@@ -5,27 +5,68 @@
  * 2.0.
  */
 
-export interface State {
+import type {
+  CoverageOverviewFilter,
+  CoverageOverviewRuleActivity,
+  CoverageOverviewRuleSource,
+} from '../../../../../common/api/detection_engine';
+import type { CoverageOverviewDashboard } from '../../../rule_management/model/coverage_overview/dashboard';
+
+export interface CoverageOverviewDashboardState {
   showExpandedCells: boolean;
+  filter: CoverageOverviewFilter;
+  isLoading: boolean;
+  data: CoverageOverviewDashboard | undefined;
 }
 
-export const initialState: State = {
-  showExpandedCells: false,
-};
-
-export interface Action {
-  type: 'setShowExpandedCells';
-  value: boolean;
-}
+export type Action =
+  | {
+      type: 'setShowExpandedCells';
+      value: boolean;
+    }
+  | {
+      type: 'setRuleActivityFilter';
+      value: CoverageOverviewRuleActivity[];
+    }
+  | {
+      type: 'setRuleSourceFilter';
+      value: CoverageOverviewRuleSource[];
+    }
+  | {
+      type: 'setRuleSearchFilter';
+      value: string;
+    };
 
 export const createCoverageOverviewDashboardReducer =
   () =>
-  (state: State, action: Action): State => {
+  (state: CoverageOverviewDashboardState, action: Action): CoverageOverviewDashboardState => {
     switch (action.type) {
       case 'setShowExpandedCells': {
         const { value } = action;
         return { ...state, showExpandedCells: value };
       }
+      case 'setRuleActivityFilter': {
```

Comment:
nit: This is one of the downsides using reducers is the action names must be synced in a number of files. It'd be nice to define supported actions like

```ts
export const SET_EXPANDED_CELLS_VISIBILITY = 'setShowExpandedCells' as const;
export const SET_FULE_ACTIVITY_FILTER = 'setRuleActivityFilter' as const;
// ...
```

then defined actions type

```ts
export type Action =
  | {
      type: typeof SET_EXPANDED_CELLS_VISIBILITY;
      value: boolean;
    }
  | {
      type: typeof SET_FULE_ACTIVITY_FILTER';
      value: CoverageOverviewRuleActivity[];
    }
  | ...
```

and reducer

```ts
     // ...
     case SET_EXPANDED_CELLS_VISIBILITY: {
        // ...
      }
      case SET_FULE_ACTIVITY_FILTER: {
        // ..
      }
```